src/Umbraco.Core/Services/PropertyValidationService.cs

```diff
@@ -291,6 +304,9 @@ private bool IsPropertyValueValid(IPropertyType propertyType, object? value, Pro
     private IDataType? GetDataType(IPropertyType propertyType)
         => _dataTypeService.GetDataType(propertyType.DataTypeId);
 
-    private IDataEditor? GetDataEditor(IPropertyType propertyType)
-        => _propertyEditors[propertyType.PropertyEditorAlias];
+    private IDataEditor GetDataEditor(IPropertyType propertyType)
+        => _propertyEditors[propertyType.PropertyEditorAlias]
+           ?? new MissingPropertyEditor(
+               propertyType.PropertyEditorAlias,
+               _dataValueEditorFactory);
```

Comment:
I don't think we need to instantiate the `MissingPropertyEditor` here, since it can't perform any validation. Instead, consider explicitly returning "no validation results" if a concrete data editor can't be found.

```suggestion
    private IDataEditor? GetDataEditor(IPropertyType propertyType)
        => _propertyEditors[propertyType.PropertyEditorAlias];
```